pkg/R/RDD.R

```diff
@@ -1348,16 +1363,16 @@ setMethod("unionRDD",
           function(x, y) {
             if (x@env$serialized == y@env$serialized) {
               jrdd <- callJMethod(getJRDD(x), "union", getJRDD(y))
-              union.rdd <- RDD(jrdd, x@env$serialized)
+              union.rdd <- RDD(jrdd, x@env$serialized, colNames = x@colNames)
             } else {
               # One of the RDDs is not serialized, we need to serialize it first.
-              if (!x@env$serialized) {
+              if (x@env$serialized == "string") {
```

Comment:
@shivaram @concretevitamin So I think the `serializedMode` problem for `UnionRDD` (and possibly `ObjectFile`) stems from the fact that `reserialize` simply calls `lapply` on the RDD, which then makes use of the fact that getJRDDs default mode is `byte`, thereby causing `getJRDD` to serialize a stringRRDD.

The issue is that `getJRDD` is now updated to dynamically alternate between `row` and `byte` by pulling serialized mode from the incoming RDD itself. This works great when you're only dealing with `row` RRDDs, but it doesn't give an easy way to force the conversion to `byte`.